store/postgres/src/detail.rs

```diff
@@ -140,8 +140,14 @@ impl TryFrom<ErrorDetail> for SubgraphError {
             block_range,
         } = value;
         let block_number = crate::block_range::first_block_in_range(&block_range).map(|n| n.into());
-        let block_ptr = block(&subgraph_id, "fatal_error", block_hash, block_number)?
-            .map(|block| block.to_ptr());
+        // In existing databases, we have errors that have a `block_range` of
+        // `UNVERSIONED_RANGE`, which leads to `None` as the block number, but
+        // has a hash. In that case, just return a block pointer of `None`
+        let block_ptr = match (&block_number, &block_hash) {
+            (Some(_), Some(_)) => block(&subgraph_id, "fatal_error", block_hash, block_number)?
```

Comment:
Going through `block` function is complicating this code, we should not convert to `BigDecimal` and construct the `EthereumBlock` directly in this branch.